sklearn/kernel_ridge/kernel_ridge.py

```diff
@@ -143,23 +123,41 @@ def fit(self, X, y=None, sample_weight=None):
         self : returns an instance of self.
         """
         # Convert data
-        X, y = check_X_y(X, y, accept_sparse=("csr", "csc"), multi_output=True,
-                         y_numeric=True)
         if sample_weight is not None and not isinstance(sample_weight, float):
             sample_weight = check_array(sample_weight, ensure_2d=False)
 
         K = self._get_kernel(X)
         alpha = np.atleast_1d(self.alpha)
 
-        ravel = False
-        if len(y.shape) == 1:
-            y = y.reshape(-1, 1)
-            ravel = True
-
         copy = self.kernel == "precomputed"
-        self.dual_coef_ = _solve_cholesky_kernel(K, y, alpha,
+
+        ravel = False
+        if self.regression is True:
+            X, y_ = check_X_y(X, y, accept_sparse=("csr", "csc"),
+                              multi_output=True, y_numeric=True)
+            if len(y_.shape) == 1:
+                y_ = y_.reshape(-1, 1)
+                ravel = True
+        else:  # Multilabel classification
+            X, y = check_X_y(X, y)
```

Comment:
Do you mean multiclass rather than multilabel? For multilabel, you need multi_output=True I think

Reply:
Done